client/service/src/chain_ops.rs

```diff
@@ -116,84 +285,50 @@ impl<
 			let client = &self.client;
 			let queue = &mut self.import_queue;
 
-			// Start by reading the number of blocks if not done so already.
-			let count = match count {
-				Some(c) => c,
-				None => {
-					let c: u64 = match Decode::decode(&mut io_reader_input) {
-						Ok(c) => c,
-						Err(err) => {
-							let err = format!("Error reading file: {}", err);
-							return std::task::Poll::Ready(Err(From::from(err)));
-						},
-					};
-					info!("📦 Importing {} blocks", c);
-					count = Some(c);
-					c
-				}
-			};
-
 			// Read blocks from the input.
-			if read_block_count < count {
-				match SignedBlock::<Self::Block>::decode(&mut io_reader_input) {
-					Ok(signed) => {
-						let (header, extrinsics) = signed.block.deconstruct();
-						let hash = header.hash();
-						// import queue handles verification and importing it into the client
-						queue.import_blocks(BlockOrigin::File, vec![
-							IncomingBlock::<Self::Block> {
-								hash,
-								header: Some(header),
-								body: Some(extrinsics),
-								justification: signed.justification,
-								origin: None,
-								allow_missing_state: false,
-								import_existing: force,
+			match block_iter.next() {
+				None => {
+					let read_block_count = block_iter.read_block_count();
+
+					info!(
+						"🎉 Imported {} blocks. Best: #{}",
```

Comment:
Printing this for each block is too much output. Printing for each 1000 blocks or once per 5 seconds would do. Ideally it should also show import speed in blocks per second. Consider reusing https://github.com/paritytech/substrate/blob/master/client/informant/src/display.rs#L110

Reply:
So maybe I misunderstood your comment, but we are not printing this for every block, only once we're done importing (block.iter() has returned None).